### PR TITLE
Try GitHub Issue Forms

### PR DESCRIPTION
**Pull request by tammullen**
## Changes proposed in this Pull Request:

Introduce GitHub issue forms to replace the existing issue templates.

[GitHub issue forms were introduced earlier this year.](https://github.blog/changelog/2021-06-23-issues-forms-beta-for-public-repositories/) They could help make it easier to triage incoming issues by improving how we capture issue details.

[More documentation on issue forms here.](https://gh-community.github.io/issue-template-feedback/welcome/)

These PR removes the following existing templates
`4-Bug-report.md`
`5-Enhancement.md`
`6-Feature-request.md`

And replaces them with
`1-bug-report.yml`
`2-enhancement.yml`

It also removes the following existing templates
`1-Security-issue.md`
`2-External-issues.md`
`3-Support.md`

And instead replaces them with external links in `config.yml` which will appear in the issue chooser.

---

**Current Issue Chooser**

Currently the Issue Chooser looks like this

Screenshot 2021-10-13 at 22 45 45

It displays all the `.md` templates in `.github/ISSUE_TEMPLATE` and pulls the description for each from the template itself.
The order of the templates is determined by the alphabetical order of the file names of the templates.
The additional `Report a security vulnerability` section at the bottom is added by default as there is a `SECURITY.md` policy in the repo

**Proposed Issue Chooser**
The proposed issue chooser will look similar to this (screenshot is a mock up from another repo , it doesn't include the security `Report a security vulnerability` section, see the individual templates and `config.yml` for the exact wording of the descriptions and links to be added)

Screenshot 2021-10-13 at 22 50 23

It displays all the `.yml` templates in `.github/ISSUE_TEMPLATE` and pulls the description for each from the template itself.
The order of the templates is determined by the alphabetical order of the file names of the templates.

After the templates there will be an additional `Report a security vulnerability` section by default as there is a `SECURITY.md` policy in the repo.

The proposed additional `config.yml` file will control the content below the templates of the issue chooser and allows us to include external links. (These links can only be displayed after the templates).

More details on configuring `conifig.yml` here https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser
 
---

**Current Bug Template**

Screenshot_2021-10-13_at_22_53_45

**Proposed Bug Template**
The proposed bug template will look similar to this (screenshot is a mock up, see `1-bug-report.yml` for the exact wording)

screencapture-github-woocommerce-woocommerce-e2e-boilerplate-issues-new-2021-10-07-20_26_44

---

**Current Enhancement Template**
Screenshot_2021-10-13_at_22_51_45

**Proposed Enhancement Template**
The proposed enhancement template will look similar to this (screenshot is a mock up, see `2-enhancement.yml` for the exact wording)

Screenshot_2021-10-13_at_22_53_22
---

## How to test the changes in this Pull Request:

- I don't know of a way to visually preview the issue chooser and the descriptions external links in `config.yml`
- To preview the issue templates view the file in the Files changed tab and GitHub will render a visual preview. (The section at the top is for the description that will be on the issue template chooser so it can be ignored).

Screenshot_2021-10-13_at_23_26_00

### CODE REVIEW

**Comment by peterfabian** [comment]
I really like these changes and don't have any specific feedback atm. 👍  I think it's great we can directly point people towards HackerOne, and also that the issue submission is much more user friendly.

File: .github/ISSUE_TEMPLATE/2-enhancement.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+labels: ["enhancement"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ### Thanks for contributing!
+
+        Please provide us with the information requested in this form. 
+
+        Make sure to look through existing issues to see whether your idea is already being discussed.
```
Comment (lsinger) [inline .github/ISSUE_TEMPLATE/2-enhancement.yml]: Could we make this [a link to existing enhancement requests](https://github.com/woocommerce/woocommerce/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+enhancement%22)?

Comment (lsinger) [reply .github/ISSUE_TEMPLATE/2-enhancement.yml]:
```suggestion
        Make sure to look through [existing issues](https://github.com/woocommerce/woocommerce/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+enhancement%22) to see whether your idea is already being discussed.
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+blank_issues_enabled: true
+contact_links:
+  - name: 🔒 Security issue
+    url: https://hackerone.com/automattic/
+    about: For security reasons, please report all security issues via HackerOne. If the issue is valid, a bug bounty will be paid out to you. Please disclose responsibly and not via GitHub (which allows for exploiting issues in the wild before the patch is released).
+  - name: ❓ Support Question
+    url: https://docs.woocommerce.com/document/woocommerce-self-service-guide/
+    about: If you have a question please see our docs or use our forums, helpdesk, or Slack community!
+  - name: WooCommerce Admin
```
Comment (lsinger) [inline .github/ISSUE_TEMPLATE/config.yml:9]: I think we've moved to a mono-repo now or will be soonish? Wouldn't the Admin and Blocks repos be obsolete then? 🤔

File: .github/ISSUE_TEMPLATE/2-enhancement.yml (hunk 1 above)
Comment (lsinger) [reply .github/ISSUE_TEMPLATE/2-enhancement.yml]: 👆 That suggestion is currently using the `type: enhancement` tag, which the PR proposes an `enhancement` label. Wouldn't we want to stick with the existing label if possible?

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (peterfabian) [reply .github/ISSUE_TEMPLATE/config.yml:9]: Admin yes, Blocks no, they stay separate for now at least.

File: .github/ISSUE_TEMPLATE/1-bug-report.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      label: Actual behavior
+      placeholder: |
+        A clear and concise description of what actually happens. Please be as descriptive as possible;
+    validations:
+      required: true
+  - type: textarea
+    id: steps
+    attributes:
+      label: Steps to reproduce
+      description: Attach screenshot(s) or recording(s) directly by dragging & dropping.
```
Comment (tammullen) [inline .github/ISSUE_TEMPLATE/1-bug-report.yml:61]: I will check if media can be included in a `textarea` or if this needs to be `markdown` instead. Might consider changing other `textarea` in the templates to `markdown`

File: .github/ISSUE_TEMPLATE/2-enhancement.yml (hunk 1 above)
Comment (tammullen) [reply .github/ISSUE_TEMPLATE/2-enhancement.yml]: Great catch, `enhancement` was the label name in the other repository I tried this in, will change it to match core label.

Comment (tammullen) [reply .github/ISSUE_TEMPLATE/2-enhancement.yml]: > Could we make this [a link to existing enhancement requests](https://github.com/woocommerce/woocommerce/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+enhancement%22)?

Yes, good idea, will do the same for the bug template and link to it to existing bugs.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (tammullen) [reply .github/ISSUE_TEMPLATE/config.yml:9]: We can submit a separate PR to remove WooCommerce Admin when that time comes.

File: .github/ISSUE_TEMPLATE/2-enhancement.yml (hunk 1 above)
Comment (tammullen) [reply .github/ISSUE_TEMPLATE/2-enhancement.yml]: Updated label and issue links

**Comment by tammullen** [comment]
Made some minor tweaks for updated documentation URLs and linked to existing issues in both templates.

@peterfabian If you are able to give it a review.

**Comment by peterfabian** [comment]
Sorry for the delay. I'll check first thing tomorrow morning.

File: .github/ISSUE_TEMPLATE/2-enhancement.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ### Thanks for contributing!
+
+        Please provide us with the information requested in this form. 
+
+        Make sure to look through [existing issues](https://github.com/woocommerce/woocommerce/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+enhancement%22) to see whether your idea is already being discussed.
+        Feel free to contribute to any existing issues.
```
Comment (peterfabian) [inline .github/ISSUE_TEMPLATE/2-enhancement.yml:14]: I wonder if we should also mention our [votes needed list](https://github.com/woocommerce/woocommerce/issues?q=is%3Aissue+sort%3Areactions-%2B1-desc+is%3Aclosed+label%3A%22votes+needed%22), wdyt?

**Review by peterfabian — DISMISSED** [review]
I'm happy with the changes, just one suggestion. Tagging @ObliviousHarmony in case folks from Atlas would like to have a look at this, too, as the newly appointed Guardians of the Monorepoxy.

File: .github/ISSUE_TEMPLATE/2-enhancement.yml (hunk 4 above)
Comment (tammullen) [reply .github/ISSUE_TEMPLATE/2-enhancement.yml:14]: Good idea, added a mention and link.

**Comment by tammullen** [comment]
@ObliviousHarmony would you be able to give this a quick final look?

**Review by ObliviousHarmony — APPROVED** [review]
I really like the look of these new issue forms, and since nothing is ever set in stone, I don't think there are any good reasons to not move forward with this.

**Pull request merged**